### PR TITLE
WIP: ci-operator/step-registry/openshift/e2e/fips: Add FIPS steps

### PR DESCRIPTION
**Pull request by wking**
And use them for master installer CI.

The implementation for the openshift-e2e-fips step is something of a union of the old-style `ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml` and the `gather-aws-console` step, which also iterates over nodes.  I'm using `read` instead of the `{1..5}` to iterate, because there may not be exactly five nodes.

The `openshift-e2e-aws-fips-default` worklow demonstrates that the default install does not enable FIPS.  I'm just using that for the installer as well; it seemed a bit heavy to add to `openshift-e2e-aws`.

WIP because we probably want to port more of the FIPS CI to steps.  I wanted to float for early feedback before putting in that effort.  CC @abhinavdahiya and @sallyom, who I've tapped as owners.  Feel free to suggest alternatives and/or additional owners.

### CODE REVIEW

File: ci-operator/step-registry/openshift/e2e/fips/openshift-e2e-fips-commands.sh [hunk 1]
```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
```
Comment (abhinavdahiya) [inline ci-operator/step-registry/openshift/e2e/fips/openshift-e2e-fips-commands.sh:1]: why this not a test in openshift-test binary?

File: ci-operator/step-registry/openshift/e2e/fips/openshift-e2e-fips-ref.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+ref:
+  as: openshift-e2e-fips
+  from: cli
+  commands: openshift-e2e-fips-commands.sh
+  env:
+  - name: TMPDIR
+    default: /tmp
+    documentation: A pathname of a directory made available for programs that need a place to create temporary files.
+  - name: FIPS_EXPECTED
```
Comment (abhinavdahiya) [inline ci-operator/step-registry/openshift/e2e/fips/openshift-e2e-fips-ref.yaml:9]: hmm.. this env is not used ..

Comment (wking) [reply ci-operator/step-registry/openshift/e2e/fips/openshift-e2e-fips-ref.yaml:9]: Oops, I need to set it in the workflow that enables FIPS...

File: ci-operator/step-registry/openshift/e2e/fips/openshift-e2e-fips-commands.sh (hunk 1 above)
Comment (wking) [reply ci-operator/step-registry/openshift/e2e/fips/openshift-e2e-fips-commands.sh:1]: I'm fine with that too.  There should be some way to figure out what the configured value should be for a node?  But I guess that wouldn't cover "installer ignores `fips` setting in the install-config".  Do we care about covering that portion?  We could probably cover it in `create manifests` integration testing in the installer.

**Comment by wking** [comment]
/hold

#10488 is dropping `fips_check` entirely, and making the argument that an MCO test is sufficient coverage.  If that holds up, we will only need the configuration step here.

**Pull request closed without merging**